tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -90,24 +146,32 @@ FORCE_INLINE void write_buffers(
         const uint32_t num_pages = command_ptr[2];
         const uint32_t page_size = command_ptr[3];
         const uint32_t dst_buf_type = command_ptr[5];
+        const uint32_t dst_page_index = command_ptr[7];
 
         uint32_t num_to_write;
         uint32_t src_addr = *reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_rd_ptr_addr(db_buf_switch)) << 4;
         uint32_t l1_consumer_fifo_limit = src_addr + consumer_cb_size - 1;
 
+        BufferType buffer_type = (BufferType)dst_buf_type;
         Buffer buffer;
-        if ((BufferType)dst_buf_type == BufferType::SYSTEM_MEMORY or not(sharded)) {
-            buffer.init((BufferType)dst_buf_type, bank_base_address, page_size);
+        if (buffer_type == BufferType::SYSTEM_MEMORY or not(sharded)) {
+            buffer.init(buffer_type, bank_base_address, page_size);
         }
         else {
             buffer.init_sharded(page_size, sharded_buffer_num_cores, bank_base_address,
                             command_ptr + COMMAND_PTR_SHARD_IDX);
         }
-        for (uint32_t id = 0; id < num_pages;) {
-            num_to_write = min(num_pages - id, producer_consumer_transfer_num_pages);
+
+        if (buffer_type == BufferType::SYSTEM_MEMORY) {
+            cq_reserve_back(num_pages * page_size);
+        }
```

Comment:
potentially we should rename this buffer type to be "completion buffer"

Reply:
makes sense because a SYSTEM_MEMORY buffer will always write into the completion queue

Reply:
won't make this change because enqueue write buffer src buff type is SYSTEM_MEMORY